client/components/jetpack/card/jetpack-product-card/use-coupon-discount.ts

```diff
@@ -0,0 +1,39 @@
+import { TERM_ANNUALLY } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
+import { INTRO_PRICING_DISCOUNT_PERCENTAGE } from 'calypso/my-sites/plans/jetpack-plans/constants';
+import { getJetpackSaleCouponDiscountRatio } from 'calypso/state/marketing/selectors';
+import type { Duration } from 'calypso/my-sites/plans/jetpack-plans/types';
+
+export default function useCouponDiscount(
+	billingTerm: Duration,
+	originalPrice?: number,
+	discountedPrice?: number
+): {
+	price?: number;
+	discount?: number;
+} {
+	const jetpackSaleDiscountRatio = useSelector( getJetpackSaleCouponDiscountRatio );
+	const introDiscountRatio =
+		billingTerm === TERM_ANNUALLY ? INTRO_PRICING_DISCOUNT_PERCENTAGE / 100 : 0;
+
+	if ( ! originalPrice ) {
+		return {};
+	}
+
+	const finalPrice =
+		Math.floor(
+			( originalPrice ?? discountedPrice ) *
```

Comment:
Seems like `originalPrice` will always be truthy, because we have a check for that a few lines above.